src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -307,7 +347,142 @@ namespace AppInstaller::CLI::Workflow
             else
             {
                 context.Add<Execution::Data::OperationReturnCode>(uninstallResult.value());
+            }
+        }
+    }
+
+#ifndef AICLI_DISABLE_TEST_HOOKS
+    std::optional<DWORD> s_EnableWindowsFeatureResult_Override{};
+
+    void TestHook_SetEnableWindowsFeatureResult_Override(std::optional<DWORD>&& result)
+    {
+        s_EnableWindowsFeatureResult_Override = std::move(result);
+    }
+
+    std::optional<DWORD> s_DoesWindowsFeatureExistResult_Override{};
+
+    void TestHook_SetDoesWindowsFeatureExistResult_Override(std::optional<DWORD>&& result)
+    {
+        s_DoesWindowsFeatureExistResult_Override = std::move(result);
+    }
+#endif
+
+    void ShellExecuteEnableWindowsFeatures(Execution::Context& context)
+    {
+        if (!Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::WindowsFeature))
+        {
+            return;
+        }
+
+        const auto& rootDependencies = context.Get<Execution::Data::Installer>()->Dependencies;
+        DWORD result = 0;
+        bool force = context.Args.Contains(Execution::Args::Type::Force);
+        bool rebootRequired = false;
+
+        rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&context, &result, &force, &rebootRequired](Dependency dependency)

```

Comment:
I think we should keep this ApplyToType part in DependenciesFLow.cpp so all dependencies related logic are in that flow. ShellExecuteInstallerHandler.cpp will only expose EnableWindowsFeature flow for enabling only 1 feature, so in the future if we are to implement winget features --enable --disable, we can reuse the same code.

Reply:
I moved the ApplyToType part back to the DependenciesFlow as suggested. I created a separate workflow just for enabling a single feature that way it can be reused in the future.